src/core/cmd.rs

```diff
@@ -4,6 +4,60 @@
 use xrl::ViewId;
 
 use std::str::FromStr;
+use serde::{Deserialize, Serialize};
+
+#[allow(non_camel_case_types)]
+#[derive(Debug, Deserialize, Serialize, PartialEq, Clone)]
+pub enum RelativeMoveDistance {
+    /// Move only one character
+    characters,
+    /// Move a line
+    lines,
+    /// Move to new word
+    words,
+    /// Move to end of word
+    word_ends,
+    /// Move to new subword
+    subwords,
+    /// Move to end of subword
+    subword_ends,
+    /// Move a page
+    pages,
+}
+
+#[allow(non_camel_case_types)]
+#[derive(Debug, Deserialize, Serialize, PartialEq, Clone)]
+pub struct RelativeMove {
+    pub by: RelativeMoveDistance,
+    pub forward: bool,
```

Comment:
I'm not sure how I feel about `forward`. On one hand, I think enums are clearer: `Direction::{Left,Up,Right,Down}` is clearer than have to think: "horizontal move + not forward => left", "vertical move + forward => down", etc. On the other, having an enum with all 4 directions here would be partially redundant with the `by` attribute, because when moving by lines, we already know the direction is either up or down for instance.

Reply:
This again comes from sublimes keymap. Thats how they specify it and I just went along. I agree that an enum would be nicer here (but I would actually just use `Direction::{Forward, Backward}`, so not to make `by` redundant). But I didn't have the time to hack that into the serde parsing.